### ISSUE TITLE
pmap doesn't work in zygote

### ISSUE THREAD

**Issue by freddycct** [FluxML/Zygote.jl#634]
```
# this demonstrates parallel backpropagation !!!

dd = Dense(10,1)

X = rand(Float32, 10, 100)
Y = rand(Float32, 1, 100)

@everywhere function loss(model, x, y)
    ŷ = model(x)
    return (y .- ŷ).^2
end

θ = params(dd)
grads = gradient(θ) do
    totalLoss = pmap(zip(eachcol(X), Y)) do xy
        x, y = xy
        loss(dd, x, y)[1]
    end
    return sqrt(sum(totalLoss))
end
Flux.Optimise.update!(optimizer, θ, grads)
```

`Compiling Tuple{typeof(Base.maptwice),Base.var"#723#725"{Channel{Any},Array{Any,1},Distributed.var"#204#207"{WorkerPool},Base.var"#716#721"{Distributed.var"#188#190"{Distributed.var"#188#189#191"{WorkerPool,var"#14#16"}}},Base.var"#722#724"{Channel{Any}}},Channel{Any},Array{Any,1},Base.Iterators.Zip{Tuple{Base.Generator{Base.OneTo{Int64},Base.var"#179#180"{Array{Float32,2}}},Array{Float32,2}}}}: try/catch is not supported.`

This worked with Tracker

**Comment by CarloLucibello** [comment]
Maybe we can simply adapt the `map` adjoint:
```julia
function ∇map(cx, f, args...)
  ys_and_backs = map((args...) -> _pullback(cx, f, args...), args...)
  if isempty(ys_and_backs)
    ys_and_backs, _ -> nothing
  else
    ys, backs = unzip(ys_and_backs)
    ys, function (Δ)
      Δf_and_args_zipped = map((f, δ) -> f(δ), backs, Δ)
      Δf_and_args = unzip(Δf_and_args_zipped)
      Δf = reduce(accum, Δf_and_args[1])
      (Δf, Δf_and_args[2:end]...)
    end
  end
end

@adjoint function map(f, args::Union{AbstractArray,Tuple}...)
  ∇map(__context__, f, args...)
end
```
but I'm not sure about it

**Comment by alok** [comment]
can this be closed since #728  is merged?

**Comment by DhairyaLGandhi** [comment]
Yes indeed, thanks!
